R/database.R

```diff
@@ -623,488 +731,539 @@ get_combined_sv = function(min_vaf=0,
+  all_cn = left_join(eg, seg_df, by = c("sample_id" = "sample_id", "region_name" = "region_name")) %>%
+    mutate(CN = replace_na(CN, 2))
+
+  cn_matrix = pivot_wider(all_cn, id_cols = "sample_id", names_from = "region_name", values_from = "CN") %>%
     column_to_rownames("sample_id")
+
   #order the regions the same way the user provided them for convenience
   return(cn_matrix[,region_names])
 }
 
-#' Get all segments for a single sample_id
+
+#' Get all segments for a single sample_id.
 #'
-#' @param sample_id The sample_id for the sample to retrieve segments for
-#' @param with_chr_prefix Set to TRUE to add a chr prefix to chromosome names
-#' @param streamlined Return a minimal output rather than full details
+#' @param this_sample_id The sample_id for the sample to retrieve segments for.
+#' @param with_chr_prefix Set to TRUE to add a chr prefix to chromosome names.
+#' @param streamlined Return a minimal output rather than full details.
+#' @param from_flatfile Resort to a flat file instead of the database for retrieving the data.
 #'
-#' @return
+#' @return A list of segments for a specific sample ID
 #' @export
 #'
 #' @examples
-get_sample_cn_segments = function(this_sample_id,with_chr_prefix=FALSE,streamlined=FALSE,from_flatfile=FALSE){
-  sample_status = get_gambl_metadata() %>%
-    dplyr::filter(sample_id==this_sample_id) %>%
+#' sample_cn_seg = get_sample_cn_segments("some-sample-id", FALSE, FALSE, FALSE)
+#'
+get_sample_cn_segments = function(this_sample_id, 
+                                  with_chr_prefix = FALSE, 
+                                  streamlined = FALSE, 
+                                  from_flatfile = FALSE){
+
+  sample_status = get_gambl_metadata() %>% 
+    dplyr::filter(sample_id == this_sample_id) %>% 
     pull(pairing_status)
+
   if(!from_flatfile){
     db = config::get("database_name")
     table_name = config::get("results_tables")$copy_number
     table_name_unmatched = config::get("results_tables")$copy_number_unmatched
-    con <- DBI::dbConnect(RMariaDB::MariaDB(), dbname = db)
+    con = DBI::dbConnect(RMariaDB::MariaDB(), dbname = db)
 
-    all_segs_matched = dplyr::tbl(con,table_name) %>%
-      dplyr::filter(ID==this_sample_id) %>%
+    all_segs_matched = dplyr::tbl(con, table_name) %>%
+      dplyr::filter(ID == this_sample_id) %>%
       as.data.frame() %>%
-      dplyr::mutate(method="battenberg")
+      dplyr::mutate(method = "battenberg")
 
     # get controlfreec segments for samples with missing battenberg results like unpaired
-    all_segs_unmatched = dplyr::tbl(con,table_name_unmatched) %>%
-      dplyr::filter(ID==this_sample_id) %>%
+    all_segs_unmatched = dplyr::tbl(con, table_name_unmatched) %>%
+      dplyr::filter(ID == this_sample_id) %>%
       as.data.frame() %>%
-      dplyr::filter(! ID %in% all_segs_matched$ID)  %>%
-      dplyr::mutate(method="controlfreec")
+      dplyr::filter(! ID %in% all_segs_matched$ID) %>%
+      dplyr::mutate(method = "controlfreec")
```

Comment:
There is also a bug here where it's returning only controlfreec results and no battenberg results somehow.

Reply:
When I run this function I get battenberg results returned. However, no controlfreec results are in the return. This is what I ran in R `sample = get_sample_cn_segments(this_sample_id = "BLGSP-71-06-00001-01A-11D")` 

Could I have the list of samples you're working with to reproduce the output you got?